src/components/EtherscanLink.tsx

```diff
@@ -29,8 +30,14 @@ function getEtherscanDomainSuffix(type: EtherscanLinkType, identifier: string):
   }
 }
 
-export const EtherscanLink: React.FC<EtherscanLinkProps> = ({ type, identifier, label, className }) => {
-  const { networkId } = useWalletConnection()
+export const EtherscanLink: React.FC<EtherscanLinkProps> = ({
+  type,
+  identifier,
+  label,
+  className,
+  networkId: netID,
+}) => {
+  const { networkId = netID } = useWalletConnection()
```

Comment:
Not for this PR, but maybe we should default to mainnet when wallet not connected?
I'd like to see the verified main contract even when I'm not connected.